### PR TITLE
Changing PlayerStates to use UUIDs instead of complete player objects

### PR DESCRIPTION
**Pull request by Maxopoly**
Fixing https://github.com/Civcraft/Bastion/issues/25

Could someone just take a second look if I missed something and merge it if it's fine?

### CODE REVIEW

File: Bastion/src/isaac/bastion/commands/PlayersStates.java [hunk 1]
```diff
@@ -16,31 +17,31 @@
...
 		public CallBack(Player player,Mode mode){
-			forPlayer=player;
+			forPlayer=player.getUniqueId();
 			forMode=mode; 
 		}
 
 		@Override
 		public void run() {
-			PlayersStates.setModeForPlayer(forPlayer, forMode);
+			PlayersStates.setModeForPlayer(Bastion.getPlugin().getServer().getPlayer(forPlayer), forMode);
```
Comment (ProgrammerDan) [inline Bastion/src/isaac/bastion/commands/PlayersStates.java]: Sanity, do an NPE catch here.

**Comment by ProgrammerDan** [comment]
Can't say if you've missed something, but besides the NPE sanity catch this looks fine.

File: Bastion/src/isaac/bastion/commands/PlayersStates.java [hunk 2]
```diff
@@ -78,7 +79,7 @@ static private void registerReturnToNormal(Player player,Mode fromMode){
 					new CallBack(player,Mode.NORMAL),
```
Comment (ProgrammerDan) [inline Bastion/src/isaac/bastion/commands/PlayersStates.java]: Why pass the player context? Just pass the Callback the UUID, callback isn't using Player, is it? No point in shoving the whole object on the callstack when the UUID will do.

**Comment by Maxopoly** [comment]
Okay, fixed the two things you found. I'll go ahead and merge it.

**Comment by ttk2** [comment]
should I stage this? has it been by Civtest?

**Comment by Maxopoly** [comment]
Yeah, I tested it, everything seemed to work and it didnt throw any exceptions, so I guess you can stage it.

**Comment by ttk2** [comment]
done.

**Comment by Maxopoly** [comment]
Oh and while doing that I realized that test.civcraft.co (the official ip) wont let anyone connect, but civcraft.co still works. No idea where to make an issue for that, just wanted to mention it. Might have been that way for a while.

**Comment by ttk2** [comment]
might not have been forwarded. @erocs

**Pull request merged**